ogc: do not create a new mouse cursor at every frame

mardy opened the pull request:
The system default cursor is not supposed to change, so we can create it just once (when the mouse gets initialized). In the polling callback we then just need to hide or show it.

The logic of showing is not changed: if the application has set a specific cursor, then it's in full control of its visibility; but if it's using the default cursor, then we show the cursor only if at least one wiimote is pointed at the screen.

rsn8887 commented:
Did you test cpuinfo detection actually works correctly with this change in Switch? You didn’t make any changes to the cpuinfo code.

uyjulian commented:
Yes, I have tested it. It allows `SDL_HasNEON` to succeed which will allow the proper audio conversion function to be used.

Cpasjuste commented:
Nice. When you say the proper conversion function, I guess you mean a neon optimised one?.
I'll take a closer look tomorrow, thanks.

uyjulian commented:
Yes. I am referring to the ones set by `SDL_ChooseAudioConverters`.   

`NEED_SCALAR_CONVERTER_FALLBACKS` is a recent addition to SDL. It is not defined when `HAVE_NEON_INTRINSICS` is defined.  
`HAVE_NEON_INTRINSICS` is defined when `__ARM_NEON` is defined.  
If `HAVE_NEON_INTRINSICS` is defined but `SDL_HasNEON` does not return true, it will cause a data abort because the audio conversion function is not set.

Cpasjuste commented:
Tested and works fine, thanks.

mardy committed:
ogc: do not create a new mouse cursor at every frame

The system default cursor is not supposed to change, so we can create it
just once (when the mouse gets initialized). In the polling callback we
then just need to hide or show it.

The logic of showing is not changed: if the application has set a
specific cursor, then it's in full control of its visibility; but if
it's using the default cursor, then we show the cursor only if at least
one wiimote is pointed at the screen.
